tests/naive.py

```diff
@@ -818,22 +839,25 @@ def update(self, t):
             D[:] = np.inf
 
         apply_exclusion_zone(D, D.shape[0] - 1, self._excl_zone, np.inf)
+        # update top-k matrix profile using newly calculated distance profile `D`
         for j in range(D.shape[0]):
-            if D[j] < self.P_[j]:
-                self.I_[j] = D.shape[0] - 1 + self._n_appended
-                self.P_[j] = D[j]
+            if D[j] < self.P_[j, -1]:
+                pos = np.searchsorted(self.P_[j], D[j], side="right")
+                self.P_[j] = np.insert(self.P_[j], pos, D[j])[:-1]
+                self.I_[j] = np.insert(
+                    self.I_[j], pos, D.shape[0] - 1 + self._n_appended
+                )[:-1]
 
-        I_last = np.argmin(D)
+        # update top-k for the last, newly-updated index
+        I_last_topk = np.argsort(D, kind="mergesort")[: self._k]
+        self.P_[-1] = D[I_last_topk]
+        self.I_[-1] = I_last_topk + self._n_appended
+        self.I_[-1][self.P_[-1] == np.inf] = -1
 
-        if np.isinf(D[I_last]):
-            self.I_[-1] = -1
-            self.P_[-1] = np.inf
-        else:
-            self.I_[-1] = I_last + self._n_appended
-            self.P_[-1] = D[I_last]
-
-        self.left_I_[-1] = I_last + self._n_appended
-        self.left_P_[-1] = D[I_last]
+        # for  last indx, the left matrix profile value is self.P_[-1, 0]
+        # and the same goes for left matrix profile index
```

Comment:
"# for **the**  last indx, the left matrix profile value is self.P_[-1, 0]"
"# and the same goes for **the** left matrix profile index"

Reply:
"index" (typo)